Sources/NIOWebSocket/NIOWebSocketClientUpgrader.swift

```diff
@@ -0,0 +1,91 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHTTP1
+
+/// A `HTTPClientProtocolUpgrader` that knows how to do the WebSocket upgrade dance.
+///
+/// This upgrader assumes that the `HTTPClientUpgradeHandler` will create and send the upgrade request. 
+/// This upgrader also assumes that the `HTTPClientUpgradeHandler` will appropriately mutate the
+/// pipeline to remove the HTTP `ChannelHandler`s.
+public final class NIOWebClientSocketUpgrader: NIOHTTPClientProtocolUpgrader {
```

Comment:
Is this a typo?
```diff
- NIOWebClientSocketUpgrader
+NIOWebSocketClientUpgrader
```

Reply:
@tanner0101 omg, yes, this is a typo :|. Mind making PR to rename (alongside a deprecated `typealias` to have the `NIOWebClientSocketUpgrader` still valid to not break SemVer).

Reply:
@tanner0101 @weissi Sorry, my bad. Definitely is a typo. I am not sure how that happened. Changing this will also require a change to the WebSocketClientDemo to use the updated class name rather than the old via the type alias. 

I am happy to make a PR for one or both of these if you would rather not do it. Let me know.

Reply:
@tigerpixel don't apologise, we all missed it :). We can land these changes in either 1 or 2 PRs, just need to make sure we retain a (deprecated) typealias so the typo'd name stays usable (for SemVer compatibility). NIO's public API breakage checker should however catch any mistakes that we might make.